packages/@aws-cdk/core/lib/lazy.ts

```diff
@@ -100,28 +140,187 @@ export interface LazyAnyValueOptions {
  * will only be calculated later, during synthesis.
  */
 export class Lazy {
+  /**
+   * Defer the calculation of a string value to synthesis time
+   *
+   * Use this if you want to render a string to a template whose actual value depends on
+   * some state mutation that may happen after the construct has been created.
+   *
+   * If you are simply looking to force a value to a `string` type and don't need
+   * the calculation to be deferred, use `Token.asString()` instead.
+   *
+   * @deprecated Use `Lazy.string()` or `Lazy.uncachedString()` instead.
+   */
   public static stringValue(producer: IStringProducer, options: LazyStringValueOptions = {}) {
-    return Token.asString(new LazyString(producer), options);
+    return Token.asString(new LazyString(producer, false), options);
+  }
+
+  /**
+   * Defer the one-time calculation of a string value to synthesis time
+   *
+   * Use this if you want to render a string to a template whose actual value depends on
+   * some state mutation that may happen after the construct has been created.
+   *
+   * If you are simply looking to force a value to a `string` type and don't need
+   * the calculation to be deferred, use `Token.asString()` instead.
+   *
+   * The inner function will only be invoked one time and cannot depend on
+   * resolution context.
+   */
+  public static string(producer: IStableStringProducer, options: LazyStringValueOptions = {}) {
```

Comment:
> Aren't we concerned about using `any`, `string`, etc for method names (from a jsii perspective?).

Seems like a valid concern to me, although I see you've done the diligence of checking through our supported languages.

Would it be safe and maybe even clearer if we add preposition to it (`asString` or `toString`) indicating which type the Lazy will evaluate into?

I'm ok with leaving as-is if you feel strongly.

Reply:
To me, `Lazy as String` or `Lazy to String` don't make sense in my head. They imply a *conversion*, which is not what's happen. What's happening is *construction* of a new kind of value.

More appropriate would be any of:

* `Lazy.aString()`
* `Lazy.newString()`
* `Lazy.makeString()`
* `Lazy.someString()`
* `Lazy.stringLazy()`
* `Lazy.lazyString()`

Reply:
Your call here. I'm not fussed.